modules/cas_cache/utils/utils_mpool.c

```diff
@@ -3,34 +3,61 @@
  * SPDX-License-Identifier: BSD-3-Clause-Clear
  */
 
-#include "ocf_env.h"
 #include "utils_mpool.h"
+#include "ocf_env.h"
+
+struct env_mpool {
+	size_t (*size_of)(uint32_t);
+		/*!< Getter for allocation size for given item count */
+
+	env_allocator *allocator[env_mpool_max];
+		/*!< OS handle to memory pool */
+
+	bool fallback;
+		/*!< Should mpool fallback to vmalloc */
 
+	int flags;
+		/*!< Allocation flags */
+};
 
-struct cas_mpool *cas_mpool_create(uint32_t hdr_size, uint32_t size, int flags,
-		int mpool_max, const char *name_perfix)
+struct env_mpool *env_mpool_create(size_t (*size_of)(uint32_t), int flags,
+		int mpool_max, bool fallback,
+		const struct env_mpool_init_limit *limits,
```

Comment:
The usage of "limits" is over complicated in my opinion. Does limits[i].size need to be one of "env_mpool_*"? If not, what is the purpose? If yes, let's just replace the limits with a simple array of integers indexed by "env_mpool_1":
     uint32_t limits[env_mpool_max]

Reply:
My intention was to specify those limits in absolute size values, not in elements count. But maybe you're right that it's overly complicated for such a simple use case, so I'll probably change it

Reply:
Changed